frontend/src/routes/Redirect.tsx

```diff
@@ -1,30 +1,26 @@
 import { Role } from '@/common';
 import { useRoleContext } from '@/providers';
-import { Navigate, Outlet } from 'react-router-dom';
+import { Navigate } from 'react-router-dom';
 import { Routes } from '.';
-import { useKeycloak } from '@react-keycloak/web';
 import { Loading } from '@/components';
 
 const Redirect = () => {
   const { roles, loading } = useRoleContext();
-  const { keycloak } = useKeycloak();
-
-  if (!keycloak.authenticated) {
-    return <Navigate to={Routes.Home} />;
-  }
   if (loading) {
     return <Loading />;
   }
-  if (roles?.includes(Role.COORDINATOR) || roles?.includes(Role.LOGISTICS)) {
+  if (
+    (roles && roles?.includes(Role.COORDINATOR)) ||
+    roles?.includes(Role.LOGISTICS)
+  ) {
     return <Navigate to={Routes.Dashboard} />;
   }
-  if (roles?.includes(Role.SUPERVISOR)) {
+  if (roles && roles?.includes(Role.SUPERVISOR)) {
```

Comment:
Was this needed? What was wrong with just roles?.includes?

Reply:
no you're right - going to merge this into the other branch and fix this + log